apps/src/music/player/ProgramSequencer.js

```diff
@@ -0,0 +1,106 @@
+// The ProgramSequencer supports the play_together and play_sequential blocks
+// in the Simple2 programming model.  It maintains a stack with context for
+// each play_together or play_sequential grouping, can return the current
+// measure whenever we want to play, and can update the playhead whenever a
+// sound has been played.  It is intended to be called by the generator functions
+// in the block model, in the order that blocks are encountered when the code
+// to be run is generated.
+
+export default class ProgramSequencer {
+  constructor() {
+    this.stack = [];
+  }
+
+  init() {
+    this.stack = [];
+  }
+
+  // Internal helper to get the entry at the top of the stack, or null
+  // if the stack is empty.
+  getCurrentStackEntry() {
+    if (this.stack.length > 0) {
+      return this.stack[this.stack.length - 1];
+    } else {
+      return null;
+    }
+  }
+
+  // Beginning of a play_sequential block.
+  playSequential() {
+    const measure =
+      this.stack.length === 0 ? 1 : this.getCurrentStackEntry().measure;
+    this.stack.push({measure: measure, together: false});
+  }
+
+  // End of a play_sequential block.
+  endSequential() {
```

Comment:
nit: I think we can consolidate `endSequential` and `endTogether`, right? The only difference seems to be the calculation of `nextMeasure`?

Reply:
Good call; factored.